src/core/helpers.cpp

```diff
@@ -49,11 +49,16 @@ getTextureData(std::string const& filename, u32& width, u32& height, bool flip)
 	auto const path = config::resources_path(filename);
 	auto const channels_nb = 4u;
 	unsigned char* image_data = stbi_load(path.c_str(), reinterpret_cast<int*>(&width), reinterpret_cast<int*>(&height), nullptr, channels_nb);
-	std::vector<unsigned char> image(width * height * channels_nb);
 	if (image_data == nullptr) {
 		LogWarning("Couldn't load or decode image file %s", path.c_str());
-		return image;
+
+		//Provide a small empty image instead in case of failure.
+		width = 16;
+		height = 16;
+		return std::vector<unsigned char>(16 * 16 * channels_nb);
```

Comment:
Other minor nitpick: please either use `width` and `height` when computing the vector size (preferred), or remove setting `width` and `height` to 16 as it has no impact.

Reply:
Used width and height in the computation as the caller expects the size to be correctly defined.